### PR TITLE
[homematic] Use .equals() to compare two Integers instead of == (code style fix)

### PR DESCRIPTION
**Pull request by pavog**
**Test Number equality with an equals() call instead of ==**

This is not so much a bug fix as a code style fix. At this point it would normally make no difference whether you use == or .equals() to compare, as the variables are compared to constants with the values 0, 1 and 2. To be on the safe side and to improve the code style in general, I would like to suggest using .equals() for comparison here.

**Let me briefly explain why I find this change important and good:**
 - With auto-boxing it is easy to make the mistake of comparing two Integer (or other subclass of java.lang.Number) objects instead of two ints.
 - The first two answers in [this stackoverflow](https://stackoverflow.com/questions/3637936/java-integer-equals-vs) explain the difference well.

This also fixes a minor typo in adresses / addresses.

### CODE REVIEW

**Review by Cossey — APPROVED** [review]
I wonder if this code style should be enforced everywhere in the future, even for primitives.

**Comment by pavog** [comment]
> I wonder if this code style should be enforced everywhere in the future, even for primitives.

Not a bad idea, because comparisons with == (especially with primitive types) can lead to very hard to detect bugs.

**Comment by Hilbrand** [comment]
In general primitives are preferred over classes. In this case if you want to do it properly you should put the constant to the left side of the equals. Because if `getNumber()` would return null it would throw a null pointer exception. (Which it would not do with the `==`  comparison. So your change might actually break the binding).

**Comment by pavog** [comment]
> In general primitives are preferred over classes. In this case if you want to do it properly you should put the constant to the left side of the equals. Because if `getNumber()` would return null it would throw a null pointer exception. (Which it would not do with the `==` comparison. So your change might actually break the binding).

Thank you for pointing this out to me. I have published an update to the pr to fix this.

File: bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java [hunk 1]
```diff
@@ -937,4 +937,4 @@ private void handleInvalidConnection(String cause) throws IOException {
             startServers();
         }
     }
-}
+}
```
Comment (Hilbrand) [inline bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java:940]: The newline at the end of the file is gone?

Comment (pavog) [reply bundles/org.openhab.binding.homematic/src/main/java/org/openhab/binding/homematic/internal/communicator/AbstractHomematicGateway.java]: Oups should be fixed now.

**Review by Hilbrand — APPROVED** [review]
Thanks. LGTM

**Pull request merged**